src/lib/chat/matrix-client.ts

```diff
@@ -512,6 +532,14 @@ export class MatrixClient implements IChatClient {
     }
   };
 
+  private onSyncStateChange = (state: SyncState, _prevState: SyncState): void => {
+    if (state === SyncState.Syncing) {
+      this.isSyncing = true;
+    } else if (state === SyncState.Stopped || state === SyncState.Error) {
+      this.isSyncing = false;
+    }
```

Comment:
What about every other possible state? We're just leaving the flag in whatever random state it was previously?